File: 3rd-sem/dsa/practicals/toh.c
#include<stdio.h>
#include<stdlib.h>

void TOH(int,char,char,char);

int main(){
    int n;
    printf("Enter the number of disks: ");
    scanf("%d",&n);
    TOH(n,'O','O','T');
}

void TOH(int n,char A, char B, char C){
    if(n>0){
        TOH(n-1,A,C,B);
        printf("Move disk to %d from %c to %c. \n",n,A,B);
        TOH(n-1,C,B,A);
    }
}

File: 4th-sem/ai/prolog/toh.pl
move(1, A, B,_) :-  
    write('move top disk from'),  
    write(A),  
    write(' to '),  
    write(B),  
    nl.  
move(N, A, B, C) :-  
    N>1,  
    M is N-1,  
    move(M, A, B, C),  
    move(1, A, B,_),  
    move(M, C, B, A). 
